test/framework/cluster.go

```diff
@@ -1172,6 +1173,20 @@ func (e *ClusterE2ETest) InstallCuratedPackageFile(packageFile, kubeconfig strin
 	})
 }
 
+// ValidateIfRegionalCuratedPackage checks if the current e2e test is a Regional Curated packages test and runs a validation.
+func (e *ClusterE2ETest) ValidateIfRegionalCuratedPackage() {
+	regionalPackagesRegex := `^.*RegionalCuratedPackages.*$`
+	if regexp.MustCompile(regionalPackagesRegex).MatchString(e.T.Name()) {
```

Comment:
Would it be better to use regex.Compile instead of MustCompile over here, MustCompile panics if the compilation fails: https://pkg.go.dev/regexp#MustCompile and if we panic in the middle of a test, we might risk losing output. In case of regex.Compile we can gracefully handle the error if any.

Reply:
What you say makes sense, but we have in fact been using `MustCompile` throughout our E2E [code](https://github.com/search?q=repo%3Aaws%2Feks-anywhere+MustCompile+path%3A%2F%5Einternal%5C%2Ftest%5C%2Fe2e%5C%2F%2F&type=code) to check for the test regex. So ideally we should change all of them, or just let this one be `MustCompile` as well.

Reply:
Right, make sense. We can address this in separate PR as well.